src/filechooser.c

```diff
@@ -563,7 +563,7 @@ handle_open (XdpImplFileChooser *object,
       if (!filters)
         {
           /* We are setting a single, unchangeable filter. */
-          gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (dialog), filter);
+          gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (dialog), g_object_ref (filter));
```

Comment:
Might be better to `g_steal_pointer()` here instead? This makes it clearer that the ownership is moved to GtkFileChooser without any annoying semantics of GInitiallyUnowned getting in the way

Reply:
Since the portal is taking ownership of the file filter through `g_autoptr`, I'd actually call `g_object_ref_sink()` unconditionally when creating the filter instance from the current_filter GVariant. This way it's explicit who owns what.

Recent versions of GLib also warn if you try to finalize a GInitiallyUnowned instance that hasn't been sunk: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2247